common/core/src/authentication_provider.ts

```diff
@@ -3,11 +3,24 @@
 
 import { TransportConfig } from './authorization';
 
+/**
+ * Designate the type of authentication used by an `AuthenticationProvider`.
+ */
 export enum AuthenticationType {
+  /**
+   * X509 authentication uses X509 certificate and key pairs.
```

Comment:
>uses X509 certificate and key pairs. [](start = 25, length = 36)

Don't really like how this reads.  The x509 certificate itself contains the public key.  The complementary private key is "securely" stored on the device.  The certificate isn't really a certificate without the public key.  Splitting hairs here.